mlflow/entities/source_type.py

```diff
@@ -8,7 +8,7 @@ class SourceType:
         "JOB": JOB,
         "PROJECT": PROJECT,
         "LOCAL": LOCAL,
-        "PIPELINE": RECIPE,
+        "RECIPE": RECIPE,
```

Comment:
Prefer to keep the `PIPELINE` key, but mapping to the same `SOURCETYPE` so that an old serialized `SourceType` can still be read and understood properly (i.e. backward-compatible).

Reply:
Above suggested approach doesn't work. The map cannot be bi-directional unless it's 1:1 mapping.
I'd prefer to not have deleted `PIPELINE` in SourceType, but since it's already done, there's no other option but also remove it from the map.

Reply:
confused - why not map `PIPELINE` to `RECIPE` and `RECIPE` to `RECIPE`? We can update the `SOURCETYPE_TO_STRING` map to special case this.

Reply:
What special case would that be? Ensure mapping `RECIPE` -> "PIPELINE"? This introduces unnecessary tech debt. I'd prefer to add `PIPELINE` back (though that's probably not a good idea at this point).

Reply:
Yes I agree, since we already made the change in MLFlow 2.0, we can just change the value and it should be fine rather than adding more tech debt. :) @dbczumar thoughts?